src/test/mocks.ts

```diff
@@ -92,7 +111,132 @@ export class MockClient extends MockWatcher implements ErdstallClient {
 		throw new Error("cannot query accounts on mock clients");
 	}
 
+	setMetadata(token: Address, id: bigint, metadata: NFTMetadata): void {
+		this.metadata.set(`${token}:${id}`, metadata);
+	}
+
+	async getNftMetadata(token: Address, id: bigint): Promise<NFTMetadata> {
```

Comment:
Don't you miss `useCache?: boolean` here now? Otherwise the mock doesn't implement the interface.

Reply:
TBH I was wondering the same initially. But truth is defining an `interface` describing some `function` with optional parameters, also makes them optional for the implementer. I expect it to work, because JS happily drops superfluous arguments.

Reply:
I see, we're in JS land, where this makes sense. If the interface check succeeds, it should be fine.